src/Http/Handler/ResponseHandlerError.php

```diff
@@ -60,15 +61,15 @@ public function execute(ResponseInterface $response)
                     $errorDescriptions[] = $error[self::FIELD_ERROR_DESCRIPTION];
                 }
 
-                throw new ApiException(
+                throw ExceptionFactory::createExceptionForResponse(
                     self::ERROR_FROM_JSON,
                     [
                         $response->getStatusCode(),
                         implode(self::SEPERATOR_ERROR, $errorDescriptions),
                     ]
                 );
             } else {
-                throw new ApiException(
+                throw ExceptionFactory::createExceptionForResponse(
                     self::ERROR_UNKNOWN,
```

Comment:
Also, please make error handling here consistent with what we have in e.g. Python :)